internal/kms/vault_sa.go

```diff
@@ -298,6 +295,16 @@ func (kms *vaultTenantSA) getToken() (string, error) {
 			kms.tenantSAName, err)
 	}
 
+	// From Kubernetes v1.24+, secret for service account tokens are not
+	// automatically created. Trying to fetch tokens from service account secret references will fail
+	// refer: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md \
+	// #no-really-you-must-read-this-before-you-upgrade-1.
+	token, err := kms.createToken(sa, c)
+	if err == nil {
+		return token, nil
+	}
+
+	log.WarningLogMsg("Failed to create SA token, err: %v. Proceeding to fetch token from secret references...", err)
```

Comment:
log this warning only once, or the log will be flooded if kubernetes doesn't support token creation.

Reply:
I can skip logging this message if that makes sense. I am not sure if there is a way to log it only once, do you have some suggestions. Please let me know.

Reply:
Thanks for your feedback. Removed logging :)